cmd/release-controller/sync_tags.go

```diff
@@ -96,12 +96,35 @@ func (c *Controller) replaceReleaseTagWithNext(release *Release, tag *imagev1.Ta
 
 func (c *Controller) removeReleaseTags(release *Release, removeTags []*imagev1.TagReference) error {
 	for _, tag := range removeTags {
-		// use delete imagestreamtag so that status tags are removed as well
-		klog.V(2).Infof("Removing release tag %s", tag.Name)
-		if err := c.imageClient.ImageStreamTags(release.Target.Namespace).Delete(context.TODO(), fmt.Sprintf("%s:%s", release.Target.Name, tag.Name), metav1.DeleteOptions{}); err != nil {
-			if !errors.IsNotFound(err) {
+		ctx := context.TODO()
+		name := fmt.Sprintf("%s:%s", release.Target.Name, tag.Name)
+		if !c.softDeleteReleaseTags {
```

Comment:
Invert these if blocks so this is `if c.softDeleteReleaseTags { ... } else { ... }`

Reply:
Updated.